### PR TITLE
create seperate loader component, fix event bug, array for btn story …

### PR DESCRIPTION
**Pull request by arslanxx**
Move Loader code to Loader component,
fix event bug
add icon along side button
array for variants

### CODE REVIEW

**Comment by arslanxx** [comment]
Screen Shot 2022-05-13 at 1 03 20 AM

File: src/components/Loader/Loader.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import StyledLoader from './styled'
+export default StyledLoader
```
Comment (bilalk1) [inline src/components/Loader/Loader.js:2]: There may be different variants of the loader component.


Please add the different sizes of the loader 
```
 size: {
    XSMALL
    DEFAULT
    MEDIUM
    LARGE
}
```
and colors of the loader.

File: src/components/Button/Button.js [hunk 2]
```diff
@@ -22,9 +23,10 @@ const Button = ({
     disabled={disabled}
     borderColor={borderColor}
     backgroundColor={backgroundColor}
-    onClick={() => handleClick}
+    onClick={handleClick}
   >
-    {loading ? <div class="loader"></div> : text}
+    { !loading && icon && <StyledIcon  src={icon} alt="icon" />}
+    { loading ? <div class="loader"></div> : text }
```
Comment (bilalk1) [inline src/components/Button/Button.js:29]: Remove the class and use style loader

**Comment by bilalk1** [comment]
Size get disturbed when you apply loader please fix it.




Screenshot 2022-05-13 at 6 52 41 PM

Screenshot 2022-05-13 at 6 52 49 PM

**Comment by arslanxx** [comment]
issues fixed.

**Pull request merged**